smart_open/s3.py

```diff
@@ -202,6 +212,7 @@ def __init__(self, bucket, key, buffer_size=DEFAULT_BUFFER_SIZE,
         self._object = s3.Object(bucket, key)
         self._raw_reader = RawReader(self._object)
         self._content_length = self._object.content_length
+        self._object.get(VersionId=self.version_id)['ContentLength']
```

Comment:
This will crash if version_id is None, won't it?

You should probably use the _get function here.